### ISSUE TITLE
[Due for payment 2025-06-12] [$250] Expense - Receipt added after creating expense changes amount & merchant field to scanning

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53535]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** V9. 0.71-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N
**Email or phone of affected tester (no customers):** N
**Issue reported by:** Applause Internal Team
**Slack UX conversation:** https://expensify.slack.com/archives/C06ML6X0W9L/p1739387850566049?thread_ts=1734034230.368829&cid=C06ML6X0W9L

## Action Performed:
1. Create a workspace if needed
2. Tap on workspace chat from workspace settings page
3. Create an expense entering an amount and merchant
4. Open the expense and upload a receipt with a different amount and merchant
5. Notice the amount and merchant change to show "Scanning..."
6. Wait a few minutes and note that the fields still show scanning
7. Go back to the workspace chat
8. Open the expense again
9. Notice the amount and merchant update to show the values from the receipt after a little while

## Expected Result:

- Ensure smart scan everything / receipt auditing is enabled by default for control workspaces only. See the related issue for that. [Receipts attached to manually created expenses do not get SmartScanned](https://github.com/Expensify/App/issues/46517#top)
- verify no fields change to “Scanning…”
- No fields are changed to the scanned values
- For workspaces without receipt auditing, verify that the attached receipt is not smart scanned and a receiptNotSmartScanned violation is displayed.
  - Update the violation copy to say something like “Receipt not smart scanned because the expense was created manually. Please verify the details. [Learn more](help-article-link) about how to enable automatic auditing for all receipts.”
  - Show the violation only to admins/approvers since the user isn’t expected to take any action
- For workspaces with receipt auditing
  - If the date doesn't match the field will show a violation
  - If the amount doesn't match and the manually entered amount is great than the scanned amount the field will show a violation

## Actual Result:

- The amount and merchant fields update to show "Scanning..." but still show the original value if you click into them
- The scanned amount and merchant overwrite the manually entered values
- The date fields shows it’s updated to the scanned date and shows a violation error “Date differs from scanned receipt”. After refreshing we see the date is actually still set to the saved date which is the date the expense was created.

## Workaround:
Unknown

## Platforms:
All
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/6de73919-6bdf-45dd-9865-b911d41cbbb8

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869484444349406830</li>
        <li>Upwork Job ID: 1869484444349406830</li>
        <li>Last Price Increase: 2025-02-27</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @OfstadC</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expense - Receipt added after creating expense changes amount & merchant field to scanning

### What is the root cause of that problem?
Whenever the receipt of the transaction is being scanned we are showing `scanning` for both amount and merchant here
https://github.com/Expensify/App/blob/64829131fe962681626b4b016140521d05f7fe52/src/components/ReportActionItem/MoneyRequestView.tsx#L213-L215
### What changes do you think we should make in order to solve the problem?

We should display scanning only if the transaction amount doesn't exist or is zero so we can update it to
```
if (TransactionUtils.hasReceipt(transaction) && TransactionUtils.isReceiptBeingScanned(transaction) && !transactionAmount) {
        merchantTitle = translate('iou.receiptStatusTitle');
        amountTitle = translate('iou.receiptStatusTitle');
    }
```
We can optionally separate the merchantTitle and only show scanning when transaction merchant is empty (or using the existing condition of `isEmptyMerchant`) or even show scanning for merchant whenever the receipt is being scanning according to the suggestions of the design team
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by OfstadC** [comment]
I haven't' had a chance to review. Will look at this this weekend or first thing Monday morning

**Comment by OfstadC** [comment]
I'm fairly certain this is intended behavior. It's returning to the original set Merchant & Amount when the receipt is done scanning. In this case the receipt did not have a merchant or amount. 

Although i'm not 100% sure what is expected if the merchant or amount was scanned off the receipt and differed from the manually entered info. 

For card expenses, historically we've taken the scanned merchant but not sure if that's the case here 🤔

**Comment by FitseTLT** [comment]
> I'm fairly certain this is intended behavior. It's returning to the original set Merchant & Amount when the receipt is done scanning. In this case the receipt did not have a merchant or amount.
> 
> Although i'm not 100% sure what is expected if the merchant or amount was scanned off the receipt and differed from the manually entered info.
> 
> For card expenses, historically we've taken the scanned merchant but not sure if that's the case here 🤔

From my observation @OfstadC The manually set amount and merchant are not overrode by scan results here is a demo: the receipt has both amount and merchant but it remained in the manual value set initially.

https://github.com/user-attachments/assets/e98cc68f-3293-4f1a-9d13-99de785d3b14

**Comment by OfstadC** [comment]
> From my observation @OfstadC The manually set amount and merchant are not overrode by scan results here is a demo: the receipt has both amount and merchant but it remained in the manual value set initially.

Interesting 🤔 .

**Comment by OfstadC** [comment]
Asked about this here - https://expensify.slack.com/archives/C06ML6X0W9L/p1734034230368829

**Comment by OfstadC** [comment]
Seems like we do want to fix this. Just waiting to confirm if this is External or Internal

**Comment by OfstadC** [comment]
I believe we do want to scan the receipt (background scan), but it shouldn't be blocking the merchant, amount when completing

**Comment by FitseTLT** [comment]
I don’t think we should display scanning because scanning never affects
manually set amount and merchant so why will hide them until the scanning
is done which doesn’t even alter them.

**Comment by FitseTLT** [comment]
@rayane-djouah I am observing that the value is updated when smart scanning is done (`modifiedAmount` and `modifiedMerchant`) but it will revert to the manual set values on page refresh (both `modifiedAmount` and `modifiedMerchant` then get reset)

https://github.com/user-attachments/assets/fc094f4f-390e-4811-8a66-0adea8a52618

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by FitseTLT** [comment]
@rayane-djouah Also have you checked https://github.com/Expensify/App/issues/53535#issuecomment-2578034351

**Comment by OfstadC** [comment]
@rayane-djouah It should be a background scan - it shouldn't be visible on the expense to the customer as "Scanning"

**Comment by FitseTLT** [comment]
> > [@rayane-djouah](https://github.com/rayane-djouah) I am observing that the value is updated when smart scanning is done (`modifiedAmount` and `modifiedMerchant`) but it will revert to the manual set values on page refresh (both `modifiedAmount` and `modifiedMerchant` then get reset)
> 
> Ah, I see
> 
>  Screen.Recording.2025-01-15.at.5.14.59.PM.1.mov 
> [@FitseTLT](https://github.com/FitseTLT) did you find a solution for this?

@rayane-djouah It is the response of the smart scanning that is pushing the result of smart scanning that should be fixed. From the FE side as long as the amount and merchant doesn't change on the BE database after smart scanning a manually set value we should stop showing scanning... as suggested by my proposal there is nothing more to add from my side.

**Comment by FitseTLT** [comment]
@rayane-djouah bump on the ^

**Comment by wildan-m** [comment]
I believe this is a Backend issue. The scanning service should send the updated scanning result to the pusher event whether it is successful or not. But it did not happen even after waiting for several minutes.

**Comment by neil-marcellini** [comment]
I'm [asking](https://expensify.slack.com/archives/C06ML6X0W9L/p1738878079905299?thread_ts=1734034230.368829&cid=C06ML6X0W9L) if we can get the issue cleaned up a bit so it's more clear, and we might have another internal engineer take it over. If not I can work on it.

**Comment by neil-marcellini** [comment]
@OfstadC thanks for updating the expected result. The first thing I notice is that the test steps seem overly involved. Why should GL and payroll code be related to this bug? First I will try a reasonable minimal set of reproduction steps and see if the issue is reproducible. Doing so clarifies the issue and saves us time when testing.

Also, I read through the [Slack conversation](https://expensify.slack.com/archives/C06ML6X0W9L/p1738878079905299?thread_ts=1734034230.368829&cid=C06ML6X0W9L) and noticed that [Jason said](https://expensify.slack.com/archives/C06ML6X0W9L/p1734035499060659?thread_ts=1734034230.368829&cid=C06ML6X0W9L) we want to show a violation if the amounts don't match. Next, I tested on OldDot to see how it works over there and what existing customers will expect.

It looks like the issue is reproducible with these minimal steps and there are some interesting UX decisions to make for NewDot. 

-------

## Action performed

1. Create a workspace if needed
2. Tap on workspace chat from workspace settings page
3. Create an expense entering an amount and merchant
4. Open the expense and upload a receipt with a different amount and merchant
5. Notice the amount and merchant change to show "Scanning..."
6. Wait a few minutes and note that the fields still show scanning
7. Go back to the workspace chat
8. Open the expense again
9. Notice the amount and merchant update to show the values from the receipt after a little while


## Expected result

- When a receipt is uploaded on an expense that already has amount and merchant fields filled, they should not update while the receipt scans in the background
- If the scanned amount and merchant don't match, a note type violation should be shown like on OldDot
  - Maybe we want to show the field violations as the error type in NewDot instead?
- It might be nice to have some other indicator that it's being scanned in the background, although that could be confusing

OldDot example:

https://github.com/user-attachments/assets/7105eaa7-dbd0-4577-b235-b85223d0430c

## Actual result
- The amount and merchant fields update to show "Scanning..." but still show the original value if you click into them
- The scanned amount and merchant overwrite the manually entered values
- The scanned date is set instead of the default current date, which seems fine, but it shows a violation "Date differs from scanned receipt" which does not appear to be true.
- If you refresh the page the date field reverts to showing the current date saved on the expense

NewDot example:

https://github.com/user-attachments/assets/6de73919-6bdf-45dd-9865-b911d41cbbb8

**Comment by neil-marcellini** [comment]
After a good discussion with Tom yesterday we got the expected UX [sorted out](https://expensify.slack.com/archives/C06ML6X0W9L/p1739387850566049?thread_ts=1734034230.368829&cid=C06ML6X0W9L). I will now update the issue to reflect that.

**Comment by neil-marcellini** [comment]
I'm also putting this on HOLD until the receipt auditing / smart scan everything logic is fixed.

- [ ] https://github.com/Expensify/App/issues/46517#top

**Comment by OfstadC** [comment]
Going to move to Weekly since this is on hold.

**Comment by neil-marcellini** [comment]
The next step is to retest and see what the current behavior looks like. I'll start on that now.

**Comment by neil-marcellini** [comment]
Here's the latest behavior.

Collect policy which doesn't have receipt auditing enabled

https://github.com/user-attachments/assets/188f0093-436b-4fd2-9f95-9ba3be149a37

Control policy with receipt auditing enabled

https://github.com/user-attachments/assets/cd6bcb6d-1ce2-44e1-b221-dce05dc7d9d9




**Current bugs**
> - Ensure smart scan everything / receipt auditing is enabled by default for control workspaces only. See the related issue for that. [Receipts attached to manually created expenses do not get SmartScanned](https://github.com/Expensify/App/issues/46517#top)

❓ I haven't checked yet but I assume this was fixed

> - verify no fields change to “Scanning…”

❌ 
> - No fields are changed to the scanned values

❌ 
> - For workspaces without receipt auditing, verify that the attached receipt is not smart scanned and a receiptNotSmartScanned violation is displayed.

❌ 
>   - Update the violation copy to say something like “Receipt not smart scanned because the expense was created manually. Please verify the details. [Learn more](help-article-link) about how to enable automatic auditing for all receipts.”

❌ 
>   - Show the violation only to admins/approvers since the user isn’t expected to take any action

❌ 
> - For workspaces with receipt auditing
>   - All fields which don’t match the scanned values show “notice” violations explaining that the field doesn’t match the scanned value 

❌ 


### Next steps

It would be great to have some help from external contributors to identify which parts of this we can fix on the frontend. When I get a chance I'll look into the backend fixes.

**Comment by FitseTLT** [comment]
@neil-marcellini Most of the bug is BE one but for the `scaning.. ` text part my proposal https://github.com/Expensify/App/issues/53535#issuecomment-2517113583 will fix it. 👍

**Comment by neil-marcellini** [comment]
I've been focused on other stuff, but this is my highest priority weekly issue. I'll set a reminder to work on it next Wednesday.

**Comment by neil-marcellini** [comment]
One small backend fix is up for review to prevent scanning receipts by default and only scan for smart scan everything policies. I'm still working on the frontend and I'm doing it myself since it's easier to test with backend changes that way.

**Comment by neil-marcellini** [comment]
May I please have a copy check for the updated violation text in the expected result?

> “Receipt not smart scanned because the expense was created manually. Please verify the details. [Learn more](https://github.com/Expensify/App/issues/help-article-link) about how to enable automatic auditing for all receipts.”

**Comment by danielrvidal** [comment]
I think I'd go with something a bit shorter: 
> Expense details and receipt added manually. Please verify the details. [Learn more](https://github.com/Expensify/App/issues/help-article-link) about automatic auditing for all receipts.

**Comment by neil-marcellini** [comment]
I plan to work on this later today if I have time. Or tomorrow.

**Comment by neil-marcellini** [comment]
I fixed the violations so they only show to admins or the manager and verified it with a manual test. Next I'll test and fix violations when automatic auditing is enabled for all receipts.

**Comment by neil-marcellini** [comment]
Based on [this message](https://expensify.slack.com/archives/C06ML6X0W9L/p1742918459057309?thread_ts=1734034230.368829&cid=C06ML6X0W9L) I updated the expected results to explain that only some fields will show violations in some cases when receipt auditing is enabled.

**Comment by neil-marcellini** [comment]
I made a lot of updates on this today and it's close to being done. However, the fields are still changing to the scanned values on control workspaces for the submitter, so I need to get that fixed still.

**Comment by neil-marcellini** [comment]
The backend PRs are up for review and the frontend PR is done. Once those go out I'll put the App PR up for review.

**Comment by neil-marcellini** [comment]
One of the backend PRs needs some tweaks, and the other one has an ongoing conversation I'm working on resolving.

**Comment by neil-marcellini** [comment]
I made some updates to both PRs and they are ready for another round of review.

**Comment by neil-marcellini** [comment]
I need to fix something with the PR so removing the reviewing label.

**Comment by neil-marcellini** [comment]
I extracted the parts to render errors with a link into the separate PR since other PRs are waiting on that. I need to get some feedback internally about an idea to fix an error with the violations.

**Comment by neil-marcellini** [comment]
[The PR](https://github.com/Expensify/App/pull/60830) is getting reviewed again to render the link errors properly. We had a bit of back and forth to confirm the UX, but hopefully it gets through review again.

**Comment by neil-marcellini** [comment]
Once this is merged I will incorporate those improvements into my original PR, and then that should complete this issue.

- [ ] https://github.com/Expensify/App/pull/61748/
- [ ] https://github.com/Expensify/App/pull/58099

**Comment by danielrvidal** [comment]
I'm going to unassign myself by let me know if I'm needed still!

**Comment by neil-marcellini** [comment]
This has taken a really long time which is really unnecessary. I should've been more proactive about keeping it updated and going forward with a solution that does not have a link in the air message.

It sounds like we're close to getting that working, but I don't want to delay it any longer so I'm going to put it up without the link rendering.

**Comment by neil-marcellini** [comment]
I got the PR updated so it doesn't render a link in the violation message. It's working well for collect workspaces, but doesn't seem to be working for Control. I'll have to look into that next.

**Comment by neil-marcellini** [comment]
I made some tweaks and pushed the app PR Forward more today.

**Comment by OfstadC** [comment]
@hoangzinh please complete BZ checklist prior to payment date

**Comment by hoangzinh** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [x] 1z. Other: It's more a feature request than a bug

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: No offending PR. It's more a feature request than a bug

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [X] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue: https://github.com/Expensify/Expensify/issues/513108

## Regression Test Proposal



### Precondition:


- Create a collect and a control workspace if needed

### Test:


Case 1: Collect workspace

1. [Owner] Sign in to an account having a collect workspace
2. [Owner] Invite a member
3. [Member] In 2nd device, log in as the member
4. [Member] Create an expense on the workspace entering an amount and merchant
5. [Member] Open the expense and upload a receipt with a different amount and merchant
6. [Member] Verify the amount and merchant do not change to show "Scanning..."
7. [Member] Submit the report if it isn't already
8. Wait a little while
9. As the workspace owner, verify that a violation appears explaining why the receipt was not smart scanned
10. As a member, verify that the violation is not displayed
11. As the owner and member, verify none of the merchant and amount fields have changed from the manually entered values


Case 2: Control workspace

1. [Owner] Sign in to an account having a control workspace
2. [Owner] Invite a member
3. [Member] In 2nd device, log in as the member
4. [Member] Create an expense so there will be a report with more than one expense
5. [Member] Create an expense on the workspace entering a large amount and any merchant
6. [Member] Open the expense and upload a receipt with a smaller amount, older date, and any merchant
7. [Member] Verify the amount and merchant do not change to show "Scanning..."
8. [Member] Verify there's no indication that the receipt is scanning on the expense, report, or workspace chat
9. Wait a little while
10. As the owner and member go to the expense and verify violations are shown on the amount and date fields

Do we agree 👍 or 👎


</details>

**Comment by OfstadC** [comment]
[Payment Summary](https://github.com/Expensify/App/issues/53535#issuecomment-2965640847) is accurate

**Comment by hoangzinh** [comment]
I requested payment in ND

**Comment by garrettmknight** [comment]
$250 approved for @hoangzinh
